src/components/Navigation/AuthenticatedNavigationView.js

```diff
@@ -43,6 +64,112 @@ export default function AuthenticatedNavigationView({ UserNavigationOptions }) {
         setMobileOpen((open) => !open);
     }, []);
 
+    function refreshTokenSetup(res) {
+        console.log('Refresh Token runs');
+        let refreshTiming = (res.tokenObj.expires_in || 3600 - 5 * 60) * 1000;
+
+        const refreshToken = async () => {
+            const newAuthRes = await res.reloadAuthResponse();
+            console.log('newAuthRes:', newAuthRes);
+            console.log('new auth Token ', newAuthRes.id.token);
+
+            setTimeout(refreshToken, refreshTiming);
+        };
+
+        setTimeout(refreshToken, refreshTiming);
+    }
+
+    useEffect(() => {
+        getCourses();
+    }, []);
+
+    useEffect(() => {
+        if (
+            email !== null &&
+            sessionStorage.getItem('google_access_token') === null
+        ) {
+            setGoogleLoginPromptOpen(true);
+        } else {
+            console.log(
+                'Use Effect: ',
+                sessionStorage.getItem('google_access_token')
+            );
+        }
+    }, [email]);
+
+    function getCourses() {
+        console.log(
+            'Courses Token: ',
+            sessionStorage.getItem('google_access_token')
+        );
+        if (
+            (gClassResp === null || gClassResp === undefined) &&
+            sessionStorage.getItem('google_access_token')
+        ) {
+            (async () => {
+                try {
+                    setGClassResp(
+                        await axios.get(
+                            'https://classroom.googleapis.com/v1/courses',
+                            {
+                                headers: {
+                                    Authorization: `Bearer ${sessionStorage.getItem(
+                                        'google_access_token'
+                                    )}`,
+                                },
+                            }
+                        )
+                    );
+                } catch (error) {
+                    console.log(error);
+                }
+            })();
+        }
+        if (google_courses === undefined || google_courses == null) {
```

Comment:
why are you setting the courses in redux if google_courses is undefined or null?